extmod/modssl_mbedtls.c

```diff
@@ -454,6 +723,36 @@ STATIC mp_uint_t socket_read(mp_obj_t o_in, void *buf, mp_uint_t size, int *errc
     } else {
         o->last_error = ret;
     }
+
+    if (
```

Comment:
This error handling code looks exactly the same as that in `socket_write` below.  So please factor them out into a helper function to reduce code duplication.